clause/clause.go

```diff
@@ -0,0 +1,19 @@
+package clause
```

Comment:
This whole package is pretty much just types and definitions of what it is to be a clause, and what information is associated with that.